src/sentry/integrations/issues.py

```diff
@@ -143,3 +146,87 @@ def sync_status_outbound(self, external_issue, is_resolved, project_id, **kwargs
         Propagate a sentry issue's status to a linked issue's status.
         """
         raise NotImplementedError
+
+    def should_resolve(self, configuration, data):
+        """
+        Given webhook data and the configuration settings, determine
+        if we should mark linked groups as resolved
+        """
+        raise NotImplementedError
+
+    def should_unresolve(self, configuration, data):
+        """
+        Given webhook data and the configuration settings, determine
+        if we should mark linked groups as unresolved
+        """
+        raise NotImplementedError
+
+    def sync_status_inbound(self, issue_key, data):
+        affected_groups = list(
+            Group.objects.get_groups_by_external_issue(
+                self.model, issue_key,
+            ).select_related('project'),
+        )
+
+        # TODO(jess): this needs to be changed to use the new model
+        project_integration_configs = {
+            pi.project_id: pi.config for pi in ProjectIntegration.objects.filter(
+                project_id__in=[g.project_id for g in affected_groups]
+            )
+        }
+
+        groups_to_resolve = []
+        groups_to_unresolve = []
+
+        for group in affected_groups:
+            project_config = project_integration_configs.get(group.project_id, {})
+            should_resolve = self.should_resolve(project_config, data)
+            should_unresolve = self.should_unresolve(project_config, data)
+            # TODO(jess): make sure config validation doesn't
+            # allow these to be the same
+            if should_resolve and should_unresolve:
+                logger.warning(
+                    'sync-config-conflict', extra={
+                        'project_id': group.project_id,
+                        'integration_id': self.model.id,
+                        'provider': self.model.get_provider(),
+                    }
+                )
+                continue
+
+            if should_unresolve:
+                groups_to_unresolve.append(group)
+            elif should_resolve:
+                groups_to_resolve.append(group)
+
+        if groups_to_resolve:
+            updated_resolve = Group.objects.filter(
```

Comment:
The two sections with `groups_to_resolve` vs `groups_to_unresolve` should probably be the same function it's pretty similar code.  Something like:
``` 
def change_status(self, groups, status, activity_type):
     updated_groups = Group.objects.filter(
                id__in=[g.id for g in groups],
      ).exclude(
                status=status,
     ).update(
                status=status,
     )
    if updated_groups:
       for group in groups:
              Activity.objects.create(
                  project=group.project,
                  group=group,
                   type=activity_type,
               )
```
so when you call it all you'll have to do is:
```
if groups_to_resolve:
   self.change_status(groups_to_resolve, GroupStatus.RESOLVED, Activity.SET_RESOLVED
``` 
And similarly with unresolved.